processor.go

```diff
@@ -118,12 +130,15 @@ func (pro *Processor) OnVote(vote *message.Vote) error {
 	// 1) collect the vote in our cache
 	err := pro.collectVote(vote)
 	if err != nil {
+		// this error contains information about the reason for not collecting a vote and is eventually handled in the caller scope
 		return rich.Errorf("could not collect vote: %w", err)
 	}
 
 	// 2) try to build proposal for next round
+	// clear from comments - create proposal on the received vote
```

Comment:
Why is the propose candidate part of the vote callback here?

Reply:
`onVote` call contains first collecting a vote in local cache records, if the node is the collector for this vote.
So when a vote is received, the node runs multiple verification steps on it - check if it is included in the graph already, not voting on a vertex that cannot be finalized, height, etc.
Potentially, if the node is the vote collector, it verifies the vote signature and if the vote has been included in its cache records.

After these checks are made, `proposeCandidate` checks if the candidate's parent already received the quorum signatures in order the vote to become a proposal for the next round - and if yes - create the candidate and sign the proposal

This step is checking if right after collecting the vote it can be pushed as a new proposal in order to process further the consensus without executing more exchanges of the vote.
And also - creating own proposal with included vote - so to gain a chance to vote and potentially get rewarded.